File: C#/10-Laço For/ContadorInteligente.cs
using System;

class Contador_inteligente
{
    static void Main(string[] args)
    {
    //O contador vai incrmentar se o segundo valor for menor que o primeiro
    //e vai decrementar caso contrario

        int i;
        int j;
        string userInput;

        Console.WriteLine("Contador Inteligente");
        Console.WriteLine("----------------------");
        Console.WriteLine("Digite um valor:");
        userInput = Console.ReadLine();
        i = Convert.ToInt32(userInput);

        Console.WriteLine("Digite outro valor:");
        userInput = Console.ReadLine();
        j = Convert.ToInt32(userInput);

        if(i < j)
        {
            for(int c = i; c <= j; c++)
            {
                Console.WriteLine(c);
            }
        }
        else if(i > j)
        {
            for(int c = i; c >= j; c--)
            {
                Console.WriteLine(c);
            }
        }
        Console.ReadLine();    
    }
}

File: Java/10-Laço For/ContadorInteligente.java
import java.util.Scanner;

class ContadorInteligente {
    public static void main(String args[]) {
        // Contador que incrementa ou decrementa dependo dos valores passados pelo
        // usuario
        int i;
        int j;
        Scanner entrada = new Scanner(System.in);

        System.out.println("Contador inteligente");
        System.out.println("---------------------");
        System.out.println("Digite um valor :");
        i = entrada.nextInt();

        System.out.println("Digite outro valor :");
        j = entrada.nextInt();

        System.out.println("Contando...");

        if (i < j) {
            for (int c = i; c <= j; c++) {
                System.out.println(c);
            }
        } else if (i > j) {
            for (int c = i; c >= j; c--) {
                System.out.println(c);
            }
        }
        entrada.close();
    }
}